### ISSUE TITLE
'HA REMOVE SERVER' doesn't work

### ISSUE THREAD

**Issue by KazuYamaguchi** [orientechnologies/orientdb#6721]
We are running two OrientDB servers.

we are trying 'HA REMOVE SERVER' command but the OrientDB doesn't became OFFLINE.

'HA REMOVE SERVER' doesn't work, as far as I see orientdb.err.

*Before running 'HA REMOVE SERVER'*

HA STATUS -servers -output=text

```java
+----------+------+------------------------------------+-----+-------------------+------------------+------------------+--------------------------+
|Name      |Status|Databases                           |Conns|StartedOn          |Binary            |HTTP              |UsedMemory                |
+----------+------+------------------------------------+-----+-------------------+------------------+------------------+--------------------------+
|orientdb1*|ONLINE|contents=ONLINE (MASTER)            |3    |2016-09-09 18:25:58|172.31.19.247:2424|172.31.19.247:2480|146.96MB/511.00MB (28.76%)|
|          |      |Test=ONLINE (MASTER)                |     |                   |                  |                  |                          |
|          |      |GratefulDeadConcerts=ONLINE (MASTER)|     |                   |                  |                  |                          |
|          |      |words=ONLINE (MASTER)               |     |                   |                  |                  |                          |
|orientdb2 |ONLINE|contents=ONLINE (MASTER)            |2    |2016-09-09 18:25:56|172.31.26.7:2424  |172.31.26.7:2480  |397.80MB/505.50MB (78.69%)|
|          |      |Test=ONLINE (MASTER)                |     |                   |                  |                  |                          |
|          |      |GratefulDeadConcerts=ONLINE (MASTER)|     |                   |                  |                  |                          |
|          |      |words=ONLINE (MASTER)               |     |                   |                  |                  |                          |
+----------+------+------------------------------------+-----+-------------------+------------------+------------------+--------------------------+
```

*After running 'HA REMOVE SERVER orientdb2'*
 
The following information is the result of the comand in the orientdb.err log.

```java:orientdb.err
2016-09-20 20:45:30:338 INFO  [orientdb1] Removing server 'orientdb2' from database configuration 'words'... [OHazelcastPlugin]
2016-09-20 20:45:30:338 INFO  [orientdb1] New distributed configuration for database: words (version=18)

CLUSTER CONFIGURATION (LEGEND: X = Owner, o = Copy)
+--------+----+-----------+----------+---------+
|        |    |           |          | MASTER  |
|        |    |           |          | ONLINE  |
+--------+----+-----------+----------+---------+
|CLUSTER |  id|writeQuorum|readQuorum|orientdb1|
+--------+----+-----------+----------+---------+
|*       |    |     1     |    1     |    X    |
|internal|   0|     1     |    1     |         |
+--------+----+-----------+----------+---------+

 [OHazelcastPlugin]
2016-09-20 20:45:30:338 INFO  [orientdb1] Saving distributed configuration file for database 'words' to: /opt/orientdb-enterprise-2.2.7/databases/words/distributed-config.json [OHazelcastPlugin]
2016-09-20 20:45:30:445 INFO  [orientdb1] Distributed servers status:

+----------+------+------------------------------------+-----+-----------------------+------------------+------------------+--------------------------+
|Name      |Status|Databases                           |Conns|StartedOn              |Binary            |HTTP              |UsedMemory                |
+----------+------+------------------------------------+-----+-----------------------+------------------+------------------+--------------------------+
|orientdb1*|ONLINE|contents=ONLINE (MASTER)            |3    |2016-09-09 18:25:58.367|172.31.19.247:2424|172.31.19.247:2480|96.02MB/510.50MB (18.81%) |
|          |      |Test=ONLINE (MASTER)                |     |                       |                  |                  |                          |
|          |      |GratefulDeadConcerts=ONLINE (MASTER)|     |                       |                  |                  |                          |
|          |      |words=ONLINE (MASTER)               |     |                       |                  |                  |                          |
|orientdb2 |ONLINE|contents=ONLINE (MASTER)            |2    |2016-09-09 18:25:56.051|172.31.26.7:2424  |172.31.26.7:2480  |420.87MB/506.00MB (83.18%)|
|          |      |Test=ONLINE (MASTER)                |     |                       |                  |                  |                          |
|          |      |GratefulDeadConcerts=ONLINE (MASTER)|     |                       |                  |                  |                          |
|          |      |words=OFFLINE (MASTER)              |     |                       |                  |                  |                          |
+----------+------+------------------------------------+-----+-----------------------+------------------+------------------+--------------------------+
 [OHazelcastPlugin]
2016-09-20 20:45:33:756 INFO  [orientdb1]<-[orientdb2] Received new status orientdb2.words=SYNCHRONIZING [OHazelcastPlugin]
2016-09-20 20:45:33:757 INFO  [orientdb1] Distributed servers status:

+----------+------+------------------------------------+-----+-----------------------+------------------+------------------+--------------------------+
|Name      |Status|Databases                           |Conns|StartedOn              |Binary            |HTTP              |UsedMemory                |
+----------+------+------------------------------------+-----+-----------------------+------------------+------------------+--------------------------+
|orientdb1*|ONLINE|contents=ONLINE (MASTER)            |3    |2016-09-09 18:25:58.367|172.31.19.247:2424|172.31.19.247:2480|96.02MB/510.50MB (18.81%) |
|          |      |Test=ONLINE (MASTER)                |     |                       |                  |                  |                          |
|          |      |GratefulDeadConcerts=ONLINE (MASTER)|     |                       |                  |                  |                          |
|          |      |words=ONLINE (MASTER)               |     |                       |                  |                  |                          |
|orientdb2 |ONLINE|contents=ONLINE (MASTER)            |2    |2016-09-09 18:25:56.051|172.31.26.7:2424  |172.31.26.7:2480  |287.46MB/506.00MB (56.81%)|
|          |      |Test=ONLINE (MASTER)                |     |                       |                  |                  |                          |
|          |      |GratefulDeadConcerts=ONLINE (MASTER)|     |                       |                  |                  |                          |
|          |      |words=SYNCHRONIZING (MASTER)        |     |                       |                  |                  |                          |
+----------+------+------------------------------------+-----+-----------------------+------------------+------------------+--------------------------+
 [OHazelcastPlugin]
2016-09-20 20:45:33:758 INFO  [orientdb1]<-[orientdb2] Updated configuration db=words [OHazelcastPlugin]
2016-09-20 20:45:33:759 INFO  [orientdb1] New distributed configuration for database: words (version=19)

CLUSTER CONFIGURATION (LEGEND: X = Owner, o = Copy)
+--------+-----------+----------+---------+-------------+
|        |           |          | MASTER  |   MASTER    |
|        |           |          | ONLINE  |SYNCHRONIZING|
+--------+-----------+----------+---------+-------------+
|CLUSTER |writeQuorum|readQuorum|orientdb1|  orientdb2  |
+--------+-----------+----------+---------+-------------+
|*       |     1     |    1     |    X    |      o      |
|internal|     1     |    1     |         |             |
+--------+-----------+----------+---------+-------------+

 [OHazelcastPlugin]
2016-09-20 20:45:33:759 INFO  [orientdb1] Saving distributed configuration file for database 'words' to: /opt/orientdb-enterprise-2.2.7/databases/words/distributed-config.json [OHazelcastPlugin]
2016-09-20 20:45:33:780 INFO  [orientdb1] Current node started as MASTER for database 'words' [OHazelcastPlugin]
2016-09-20 20:45:33:800 INFO  [orientdb1] Received updated status orientdb2.words=SYNCHRONIZING [OHazelcastPlugin]
2016-09-20 20:45:33:816 INFO  [orientdb1]->[orientdb2] Deploying database words... [OSyncDatabaseTask]
2016-09-20 20:45:33:816 INFO  [orientdb1]->[orientdb2] Reusing last backup of database 'words' in directory: /tmp/orientdb/backup_words.zip... [OSyncDatabaseTask]
2016-09-20 20:45:33:819 INFO  [orientdb1]->[orientdb2] - transferring chunk #1 offset=0 size=5340364... [OSyncDatabaseTask]
2016-09-20 20:45:33:821 INFO  [orientdb1] Received updated status orientdb1.words=SYNCHRONIZING [OHazelcastPlugin]
2016-09-20 20:45:33:822 INFO  [orientdb1] Distributed servers status:

+----------+------+------------------------------------+-----+-----------------------+------------------+------------------+--------------------------+
|Name      |Status|Databases                           |Conns|StartedOn              |Binary            |HTTP              |UsedMemory                |
+----------+------+------------------------------------+-----+-----------------------+------------------+------------------+--------------------------+
|orientdb1*|ONLINE|contents=ONLINE (MASTER)            |3    |2016-09-09 18:25:58.367|172.31.19.247:2424|172.31.19.247:2480|96.02MB/510.50MB (18.81%) |
|          |      |Test=ONLINE (MASTER)                |     |                       |                  |                  |                          |
|          |      |GratefulDeadConcerts=ONLINE (MASTER)|     |                       |                  |                  |                          |
|          |      |words=SYNCHRONIZING (MASTER)        |     |                       |                  |                  |                          |
|orientdb2 |ONLINE|contents=ONLINE (MASTER)            |2    |2016-09-09 18:25:56.051|172.31.26.7:2424  |172.31.26.7:2480  |287.46MB/506.00MB (56.81%)|
|          |      |Test=ONLINE (MASTER)                |     |                       |                  |                  |                          |
|          |      |GratefulDeadConcerts=ONLINE (MASTER)|     |                       |                  |                  |                          |
|          |      |words=SYNCHRONIZING (MASTER)        |     |                       |                  |                  |                          |
+----------+------+------------------------------------+-----+-----------------------+------------------+------------------+--------------------------+
 [OHazelcastPlugin]
2016-09-20 20:45:33:837 INFO  [orientdb1] Received updated status orientdb1.words=ONLINE [OHazelcastPlugin]
2016-09-20 20:45:33:839 INFO  [orientdb1]->[orientdb2] Deploy database task completed [OSyncDatabaseTask]
2016-09-20 20:45:33:840 INFO  [orientdb1] Distributed servers status:

+----------+------+------------------------------------+-----+-----------------------+------------------+------------------+--------------------------+
|Name      |Status|Databases                           |Conns|StartedOn              |Binary            |HTTP              |UsedMemory                |
+----------+------+------------------------------------+-----+-----------------------+------------------+------------------+--------------------------+
|orientdb1*|ONLINE|contents=ONLINE (MASTER)            |3    |2016-09-09 18:25:58.367|172.31.19.247:2424|172.31.19.247:2480|96.02MB/510.50MB (18.81%) |
|          |      |Test=ONLINE (MASTER)                |     |                       |                  |                  |                          |
|          |      |GratefulDeadConcerts=ONLINE (MASTER)|     |                       |                  |                  |                          |
|          |      |words=ONLINE (MASTER)               |     |                       |                  |                  |                          |
|orientdb2 |ONLINE|contents=ONLINE (MASTER)            |2    |2016-09-09 18:25:56.051|172.31.26.7:2424  |172.31.26.7:2480  |287.46MB/506.00MB (56.81%)|
|          |      |Test=ONLINE (MASTER)                |     |                       |                  |                  |                          |
|          |      |GratefulDeadConcerts=ONLINE (MASTER)|     |                       |                  |                  |                          |
|          |      |words=SYNCHRONIZING (MASTER)        |     |                       |                  |                  |                          |
+----------+------+------------------------------------+-----+-----------------------+------------------+------------------+--------------------------+
 [OHazelcastPlugin]
2016-09-20 20:45:34:326 INFO  [orientdb1]<-[orientdb2] Received updated status orientdb2.words=ONLINE [OHazelcastPlugin]
2016-09-20 20:45:34:326 INFO  [orientdb1] Distributed servers status:

+----------+------+------------------------------------+-----+-----------------------+------------------+------------------+--------------------------+
|Name      |Status|Databases                           |Conns|StartedOn              |Binary            |HTTP              |UsedMemory                |
+----------+------+------------------------------------+-----+-----------------------+------------------+------------------+--------------------------+
|orientdb1*|ONLINE|contents=ONLINE (MASTER)            |3    |2016-09-09 18:25:58.367|172.31.19.247:2424|172.31.19.247:2480|96.02MB/510.50MB (18.81%) |
|          |      |Test=ONLINE (MASTER)                |     |                       |                  |                  |                          |
|          |      |GratefulDeadConcerts=ONLINE (MASTER)|     |                       |                  |                  |                          |
|          |      |words=ONLINE (MASTER)               |     |                       |                  |                  |                          |
|orientdb2 |ONLINE|contents=ONLINE (MASTER)            |2    |2016-09-09 18:25:56.051|172.31.26.7:2424  |172.31.26.7:2480  |287.46MB/506.00MB (56.81%)|
|          |      |Test=ONLINE (MASTER)                |     |                       |                  |                  |                          |
|          |      |GratefulDeadConcerts=ONLINE (MASTER)|     |                       |                  |                  |                          |
|          |      |words=ONLINE (MASTER)               |     |                       |                  |                  |                          |
+----------+------+------------------------------------+-----+-----------------------+------------------+------------------+--------------------------+
 [OHazelcastPlugin]
2016-09-20 20:45:34:328 INFO  [orientdb1]<-[orientdb2] Updated configuration db=words [OHazelcastPlugin]
2016-09-20 20:45:34:329 INFO  [orientdb1] New distributed configuration for database: words (version=33)

CLUSTER CONFIGURATION (LEGEND: X = Owner, o = Copy)
+-----------+-----------+----------+---------+---------+
|           |           |          | MASTER  | MASTER  |
|           |           |          | ONLINE  | ONLINE  |
+-----------+-----------+----------+---------+---------+
|CLUSTER    |writeQuorum|readQuorum|orientdb1|orientdb2|
+-----------+-----------+----------+---------+---------+
|*          |     1     |    1     |    X    |    o    |
|_studio    |     1     |    1     |    o    |    X    |
|distance_0 |     1     |    1     |    o    |    X    |
|e_0        |     1     |    1     |    o    |    X    |
|internal   |     1     |    1     |         |         |
|ofunction_0|     1     |    1     |    o    |    X    |
|orids_0    |     1     |    1     |    o    |    X    |
|orole_0    |     1     |    1     |    o    |    X    |
|oschedule_0|     1     |    1     |    o    |    X    |
|ouser_0    |     1     |    1     |    o    |    X    |
|relation2_0|     1     |    1     |    o    |    X    |
|relation3_0|     1     |    1     |    o    |    X    |
|v_0        |     1     |    1     |    o    |    X    |
|word2_0    |     1     |    1     |    o    |    X    |
|word3_0    |     1     |    1     |    o    |    X    |
|word_0     |     1     |    1     |    o    |    X    |
+-----------+-----------+----------+---------+---------+

 [OHazelcastPlugin]
2016-09-20 20:45:34:329 INFO  [orientdb1] Saving distributed configuration file for database 'words' to: /opt/orientdb-enterprise-2.2.7/databases/words/distributed-config.json [OHazelcastPlugin]
2016-09-20 20:45:34:329 INFO  [orientdb1] Current node started as MASTER for database 'words' [OHazelcastPlugin]

```

Does 'HA REMOVE SERVER' works but immediately the server changed OFFLINE goes back ONLINE?

**Comment by lvca** [comment]
From the logs it looks like you're right: the database is automatically re-installed after the REMOVE. I'm trying to reproduce it.

**Comment by KazuYamaguchi** [comment]
Hi @lvca 
If you need more information, let me know.
Thanks
kazunobu

**Comment by lvca** [comment]
Reproduced, working toward for a fix.

**Comment by lvca** [comment]
Fixed in 2.2.x branch. In the EE I've also added 3 new commands to have more control of the replication:
```
HA START REPLICATION <server> [-mode <FULL|DELTA>]
HA STOP REPLICATION <server> [-dropDatabase]
HA SET [DBSTATUS <server>=<status>]
```

Look at http://www.prjhub.com/#/issues/7659.
